flink-runtime/src/test/java/org/apache/flink/runtime/leaderelection/ZooKeeperLeaderElectionTest.java

```diff
@@ -409,20 +408,19 @@ void testExceptionForwarding() throws Exception {
 
             final TestingLeaderElectionListener electionEventHandler =
                     new TestingLeaderElectionListener();
-            final TestingFatalErrorHandler fatalErrorHandler =
-                    testingFatalErrorHandlerResource.getTestingFatalErrorHandler();
             try (ZooKeeperLeaderElectionDriver leaderElectionDriver =
-                    createAndInitLeaderElectionDriver(
-                            client, electionEventHandler, fatalErrorHandler)) {
+                    createAndInitLeaderElectionDriver(client, electionEventHandler)) {
 
                 electionEventHandler.await(LeaderElectionEvent.IsLeaderEvent.class);
 
                 leaderElectionDriver.publishLeaderInformation(
                         CONTENDER_ID, LeaderInformation.known(UUID.randomUUID(), "some-address"));
 
-                assertThat(fatalErrorHandler.getErrorFuture()).isCompletedWithValue(testException);
+                final LeaderElectionEvent.ErrorEvent errorEvent =
+                        electionEventHandler.await(LeaderElectionEvent.ErrorEvent.class);
+                assertThat(errorEvent.getError()).isEqualTo(testException);
             } finally {
-                fatalErrorHandler.clearError();
+                electionEventHandler.throwIfErrorEventHappened();
```

Comment:
This change will cause all the test methods to change there signature from throwing `Throwable` instead of `Exception`. But that's what we want because we also would want to forward Errors, not only Exceptions.

Reply:
> But that's what we want because we also would want to forward Errors, not only Exceptions.

Could we just wrap everything into an exception? :thinking:

Reply:
Or even better, an unchecked exception?

Reply:
I pushed the changes related to your recommendation.